### PR TITLE
feat: Initial SPI components deployment

### PR DESCRIPTION
**Pull request by skabashnyuk**
## Details 
This is a initial deployment if SPI components
 - `service-provider-integration-operator` https://github.com/redhat-appstudio/service-provider-integration-operator
 - `service-provider-integration-oauth` https://github.com/redhat-appstudio/service-provider-integration-oauth

Знімок екрана 2021-12-30 о 10 42 52
Знімок екрана 2021-12-30 о 12 10 39
Знімок екрана 2021-12-30 о 12 11 10

Now both components are at the initial stage of **SPI V2** implementation  https://docs.google.com/document/u/1/d/1Agp8WMn5uSyzJBin1JfbgCIlKbGt5ziHV_aF2wT5eKc/edit?usp=drive_web&ouid=110694458140179268052
 - `service-provider-integration-operator` provides CRDs https://docs.google.com/document/d/1ymo3DgBKdKAq0z1SNgM6PcOqpJyi4EDjO0HVBWPrYn4/edit
 - `service-provider-integration-oauth` implemented OAuth2 flow for Github.

## NOTICE
In this pr OAuth service configuration is represented as ` spi-oauth-config-xxx` and `spi-shared-secret` contains dummy data. In the future, it has to be replaced with some production-grade provider of secrets. See also https://github.com/redhat-appstudio/infra-deployments/pull/46

### CODE REVIEW

**Review by jduimovich — COMMENTED** [review]
The name seems to be in a strange font.  Unclear this is intentional or whether it will even cause issues so this is an FYI  See this
 image

**Comment by skabashnyuk** [comment]
> The name seems to be in a strange font. Unclear this is intentional or whether it will even cause issues so this is an FYI See this image

@jduimovich I'm still experimenting with vault deployment. After a few tries, I've cleaned up these charts.

**Comment by skabashnyuk** [comment]
I've added a **draft** mark for this pr. The reason is that this pr has to be reworked in the section of operator deployment. It has to use regular deployment instead of OLM

**Comment by skabashnyuk** [comment]
@sbose78 @jduimovich pr un-drafted and ready to review. The thing that I'm not sure about is about the cert-manager. Maybe it's better to rework it this way https://docs.openshift.com/container-platform/4.6/security/certificates/service-serving-certificate.html#add-service-certificate-validating-webhook_service-serving-certificate

**Review by sbose78 — COMMENTED** [review]
Could you check if you've done the equivalent of https://github.com/redhat-appstudio/application-service/pull/23 ?

**Comment by skabashnyuk** [comment]
>Could you check if you've done the equivalent of redhat-appstudio/application-service#23 ?

@sbose78 done in https://github.com/redhat-appstudio/service-provider-integration-operator/pull/51

**Comment by sbose78** [comment]
Thanks, I will review!

File: components/spi/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+resources:
+  - https://github.com/redhat-appstudio/service-provider-integration-operator/config/default?ref=d7a6475c6d948452cdc00941616024c9ef6607cc
```
Comment (sbose78) [inline components/spi/kustomization.yaml:2]: We'll need https://github.com/redhat-appstudio/service-provider-integration-operator/commit/4f8d0b4d4e9980518f9a759f7683639e025bc32c ?

**Review by sbose78 — CHANGES_REQUESTED** [review]

**Review by sbose78 — APPROVED** [review]

**Comment by flacatus** [comment]
I see the e2e are failing with an ?unexpected? error in has component:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/29/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-deployment/1481492363904815104/artifacts/appstudio-e2e-deployment/gather-extra/artifacts/pods/application-service_application-service-controller-manager-64bbb845f-rchss_manager.log

**Pull request merged**